crossplane/function/resource.py

```diff
@@ -69,10 +70,7 @@ def struct_to_dict(s: structpb.Struct) -> dict:
     protobuf struct. This function makes it possible to convert resources to a
     dictionary.
     """
-    return {
-        k: (struct_to_dict(v) if isinstance(v, structpb.Struct) else v)
-        for k, v in s.items()
-    }
+    return json_format.MessageToDict(s, preserving_proto_field_name=True)
```

Comment:
Nice. Can you update `dict_to_struct` as well per https://github.com/crossplane/function-sdk-python/issues/108?

Reply:
done!

Reply:
but CI is failing :rage4:

Reply:
fixed